GpxLocationManager/GpxLocationManager.swift

```diff
@@ -154,7 +154,17 @@ open class GpxLocationManager {
             currentLocation = CLLocation(coordinate: lastLoc.coordinate, altitude: lastLoc.altitude, horizontalAccuracy: lastLoc.horizontalAccuracy, verticalAccuracy: lastLoc.verticalAccuracy, course: lastLoc.course, speed: 0.0, timestamp: startDate.addingTimeInterval(timeIntervalSinceStart))
           } else {
             currentLocation = self.locations[currentIndex]
-            currentLocation = CLLocation(coordinate: currentLocation.coordinate, altitude: currentLocation.altitude, horizontalAccuracy: currentLocation.horizontalAccuracy, verticalAccuracy: currentLocation.verticalAccuracy, course: currentLocation.course, speed: currentLocation.speed, timestamp: currentLocation.timestamp.addingTimeInterval((routeDuration + TimeInterval(1.0)) * TimeInterval(loopsCompleted)))
+
+            let timeInterval = (routeDuration + TimeInterval(1.0)) * TimeInterval(loopsCompleted) + self.secondLength // true of the previous and the current span
+            let speed: CLLocationSpeed
+            if currentLocation.speed != -1 && currentLocation.speed != Double.infinity {
+                speed = currentLocation.speed
+            } else if self.locations.count >= currentIndex {
```

Comment:
Should this be a strictly less than? i.e. `self.locations.count > currentIndex`?

Reply:
Or really `self.locations.count - 1 > currentIndex`